subsys/net/lib/nrf_cloud/src/nrf_cloud_codec_internal.c

```diff
@@ -496,7 +496,11 @@ int nrf_cloud_state_encode(uint32_t reported_state, const bool update_desired_to
 {
 	__ASSERT_NO_MSG(output != NULL);
 
-	char *buffer;
+	if ((reported_state != STATE_UA_PIN_WAIT) && (reported_state != STATE_UA_PIN_COMPLETE)) {
```

Comment:
A comment here explaining that `if ((reported_state != STATE_UA_PIN_WAIT) && (reported_state != STATE_UA_PIN_COMPLETE))` essentially corresponds to a disconnected / failed connection state would be helpful. At least, I think it does? Or is the state straight up just invalid?